### ISSUE TITLE
Problem using generators with spell-spec specs

### ISSUE THREAD

**Issue by kthu** [bhauman/spell-spec#1]
I tried dropping in spell-spec's keys macro as a replacement for spec/keys for some maps that are user provided. 

I have some tests that are exercising the functions that take or return these kinds of maps, and those test broke when using spell-spec.

Here is an example:

```clojure
user=> (require '[clojure.spec.alpha :as s])
nil
user=> (require '[spell-spec.alpha :as spell])
nil
user=> (s/def ::baz string?)
:user/baz
user=> (s/def ::foo (s/keys :req-un [::baz]))
:user/foo
user=> (s/def ::bar (spell/keys :req-un [::baz]))
:user/bar
user=> (s/gen ::foo)
#clojure.test.check.generators.Generator{:gen #object[clojure.test.check.generators$such_that$fn__6411 0x4a2929a4 "clojure.test.check.generators$such_that$fn__6411@4a2929a4"]}
user=> (s/gen ::bar)
ExceptionInfo Unable to construct gen at: [0] for: clojure.lang.LazySeq@fa039170  clojure.core/ex-info (core.clj:4739)
user=> 
```

Giving generators explicitly works around the problem:

```clojure
user=> (s/def ::bar (s/with-gen (spell/keys :req-un [::baz])
  #_=>        #(s/gen (s/keys :req-un [::baz]))))
:user/bar
user=>(s/gen ::bar)
#clojure.test.check.generators.Generator{:gen #object[clojure.test.check.generators$such_that$fn__6411 0x7132a9dc "clojure.test.check.generators$such_that$fn__6411@7132a9dc"]}
user=> 
```

but it quickly gets a bit verbose.

**Comment by bhauman** [comment]
Sorry I should have tested this.  I will have this fixed by Monday.

**Comment by bhauman** [comment]
fixed and deployed to 0.1.1

**Comment by kthu** [comment]
Thank you! Works like a charm :)
